Sofa/Component/Mass/tests/DiagonalMass_test.cpp

```diff
@@ -984,6 +985,97 @@ class DiagonalMass_test : public BaseTest
         EXPECT_EQ(vMasses.size(), 0);
         EXPECT_NEAR(mass->getTotalMass(), 0, 1e-4);
     }
+
+    static Node::SPtr generateRigidScene()
+    {
+        static const string scene =
+        "<?xml version='1.0' ?>"
+        "<Node name='root' dt='0.01' gravity='0 0 0'>"
+        "    <RequiredPlugin name='Sofa.Component.Mass'/>"
+        "    <RequiredPlugin name='Sofa.Component.StateContainer'/>"
+        "    <RequiredPlugin name='Sofa.Component.Topology.Container.Grid'/>"
+        "    <RequiredPlugin name='Sofa.Component.Visual'/>"
+        "    <RequiredPlugin name='Sofa.Component.ODESolver.Backward'/>"
+        "    <RequiredPlugin name='Sofa.Component.LinearSolver.Direct'/>"
+        "    <RequiredPlugin name='Sofa.Component.Engine.Select'/>"
+        "    <RequiredPlugin name='Sofa.Component.Constraint.Projective'/>"
+        "    <RequiredPlugin name='Sofa.Component.SolidMechanics.FEM.Elastic'/>"
+        "    <RequiredPlugin name='Sofa.Component.MechanicalLoad'/>"
+        "    <DefaultAnimationLoop />"
+        "    <EulerImplicitSolver rayleighStiffness='0.'  rayleighMass='0.0'/>"
+        "    <SparseLDLSolver applyPermutation='false' template='CompressedRowSparseMatrixMat3x3d'/>"
+        "    <Node name='Aligned' >"
+        "        <MechanicalObject  name='Mstate1' template='Rigid3' position='0 0 0 0 0 0 1' showObject='true' showObjectScale='0.1'/>"
```

Comment:
```suggestion
        "    <SparseLDLSolver applyPermutation='false' template='CompressedRowSparseMatrixd'/>"
        "    <Node name='Aligned' >"
        "        <MechanicalObject  name='Mstate1' template='Rigid3' position='0 0 0 0 0 0 1' showObject='true' showObjectScale='0.1'/>"
```

`CompressedRowSparseMatrixMat3x3d` for `SparseLDLSolver` should be used only with MechanicalObject templated on Vec3 no ?

Reply:
and applyPermutation='false' is deprecated as well

Reply:
Anyway, even with the corrected template, it still fails.

Reply:
You are right ! As you might guess, this comes from an old scene of mine and I didn't check the solver template...